hazelcast/src/main/java/com/hazelcast/internal/plugin/multicast/MulticastDiscoveryStrategy.java

```
@@ -0,0 +1,134 @@
+import com.hazelcast.spi.discovery.SimpleDiscoveryNode;
+import com.hazelcast.spi.partitiongroup.PartitionGroupStrategy;
+
+import java.net.InetAddress;
+import java.net.MulticastSocket;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class MulticastDiscoveryStrategy implements DiscoveryStrategy {
+
+    private DiscoveryNode discoveryNode;
+    private MulticastSocket multicastSocket;
+    Thread t;
+    private Map<String, Comparable> properties;
+    private static final int DATA_OUTPUT_BUFFER_SIZE = 1024;
+    boolean isClient;
+
+    private MulticastDiscoveryReceiver multicastDiscoveryReceiver;
+    private MulticastDiscoverySender multicastDiscoverySender;
+
+    public MulticastDiscoveryStrategy(DiscoveryNode discoveryNode, Map<String, Comparable> properties) {
+        this.discoveryNode = discoveryNode;
+        this.properties = properties;
+
+    }
+
+    private void initializeMulticastSocket() {
+        try {
+            int port = getOrDefault(MulticastProperties.PORT, 54327);
+            String group = getOrDefault(MulticastProperties.GROUP, "224.2.2.3");
+            multicastSocket = new MulticastSocket(port);
+            multicastSocket.setReuseAddress(true);
+            multicastSocket.setTimeToLive(255);
+            multicastSocket.setReceiveBufferSize(64 * DATA_OUTPUT_BUFFER_SIZE);
+            multicastSocket.setSendBufferSize(64 * DATA_OUTPUT_BUFFER_SIZE);
+            multicastSocket.setSoTimeout(3000);
+            multicastSocket.joinGroup(InetAddress.getByName(group));
+            multicastDiscoverySender = new MulticastDiscoverySender(discoveryNode, multicastSocket);
+            multicastDiscoveryReceiver = new MulticastDiscoveryReceiver(multicastSocket);
+            if (discoveryNode != null) {
+                isClient = false;
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Probably you want to propagate this exception.